### ISSUE TITLE
Access to parsed Tokens data on mapping error

### ISSUE THREAD

**Issue by Bertrand-Sg** [bytefish/TinyCsvParser#36]
Thanks for the work done on this library! 
I've been looking at a few options to parse big csv files (2Gb+) and the performance of your implementation is remarkable. 

I'm facing an issue now that could be a potential feature enhancement. At the end of the 2Gb files that I am parsing there are "footer" rows that can't be Mapped to `TEntity` but they contain meaningful information on the file. 
Parsing the big file once being already time consuming, I'd like to avoid an other full parse.
These lines are being mentioned in the Error collection but their Value is the same for all the footer lines [Column 2 is Out Of Range] and I can't access the Raw data from there.

One possible fix would be to add a property with the raw data `UnmappedRow` to all error lines. I can think of a few other scenarios where a look through the raw data would be handy when an error occurs.

```cs 
    public class CsvMappingError
    {
        public int ColumnIndex { get; set; }
        public string Value { get; set; }
        public string UnmappedRow { get; set; }
        public override string ToString()
        {
            return $"CsvMappingError (ColumnIndex = {ColumnIndex}, Value = {Value}, UnmappedRow = {UnmappedRow})";
        }
    }
```

And the `CsvMapping.cs` could add the raw data to the the `Map()` property. Something like:

```csharp
Error = new CsvMappingError()
{
     ColumnIndex = columnIndex,
     Value = $"Column {columnIndex} is Out Of Range",
     UnmappedRow = string.Join("|", values.Tokens)
}
```

Alternatively, an UnmappedTokens porperty would also be helpful.
```csharp
public IEnumerable<string> UnmappedTokens { get; set; }
```
Thank you

**Comment by bytefish** [comment]
@Bertrand-Sg Good idea! 🤓 It would be great, if you make a Pull Request. I am happy to accept it! 🤝

**Comment by Bertrand-Sg** [comment]
Thanks for the quick reply!
I've made the change (https://github.com/Bertrand-Sg/TinyCsvParser) but wasn't able to run Unit test considering your multi-target TargetFramework setup. TinyCsvParser (.NetFramework4.5 & .NetStandard 1.3) with Test (.NetCore 1.1.2). I've tried VS, Resharper, NUnit, msbuid test... without success. What am I missing? Can you share a bit about your setup to run the tests?

**Comment by bytefish** [comment]
Can you send me, what's the error message? It's basically using NUnit3 and the NUnit3 Testadapter package from NuGet should be installed. Both are referenced in the Test Project, so there shouldn't be a problem.

Could you share the error message?

**Comment by bytefish** [comment]
@Bertrand-Sg What Visual Studio Version are you using?

**Comment by Bertrand-Sg** [comment]
I'm using VS Enterprise 2017. I found the issue, I couldn't run the test because I was missing the .NetCore SDK 1.1.2 and the Installer starts at v1.1.6 (https://dotnet.microsoft.com/download/dotnet-core/1.1). I was able to run them in command line and validate that all tests where passing

**Comment by bytefish** [comment]
This is weird. I will adjust the netcoreapp version, so it also runs with the VS 2017 default installation. Maybe it worked for me, because I have an old .NET Core 1.1 installation on my machine.

**Comment by bytefish** [comment]
I have merged the Pull Request. Great, that you also wrote a Unit Test! It may take some time until the evening to build the new NuGet package (it's 6 am here). Good job. 🤜🤛

**Comment by Bertrand-Sg** [comment]
Thanks for accepting the change and your quick replies!

**Comment by bytefish** [comment]
@Bertrand-Sg I have fixed the Unit Tests to target a more recent netcoreapp version and released a Version 2.2.0 to NuGet. Feel free to reopen the Ticket, if there are any problems. Thanks again for your contribution! 👍
